Std/Tactic/Ext.lean

```diff
@@ -106,25 +127,65 @@ def applyExtLemma (goal : MVarId) : MetaM (List MVarId) := goal.withContext do
         let c ← mkConstWithFreshMVarLevels lem.declName
         let (_, _, declTy) ← withDefault <| forallMetaTelescopeReducing (← inferType c)
         guard (← isDefEq tgt declTy)
+      if tactic.ext.trace.get (← getOptions) then
+        let cmd ←`(tactic| apply $(mkIdent (← unresolveNameGlobal lem.declName)))
+        usedExtTactics.modify (·.push cmd)
       -- We use `newGoals := .all` as this is
       -- more useful in practice with dependently typed arguments of `@[ext]` lemmas.
       return ← goal.apply (cfg := { newGoals := .all }) (← mkConstWithFreshMVarLevels lem.declName)
     catch _ => s.restore
+  -- Fall back to try all `ext`-lemmas in case of failure
+  if tactic.ext.force.get (← getOptions) then
+    for (key, lemmas) in ← getAllExtLemmas do
+      -- TODO: `if key == ty then continue` to avoid going over the same key twice.
+      -- As far as I see, we need more API from
+      -- [Lean.Meta.DiscrTree] exposed, but everything there is private.
+      -- The first projection of `getMatchCore` for example
```

Comment:
This is a bit unfortunate, we might iterate again over the "good" ext lemmas we already tried. To avoid it, we need to compare `ty : Expr` with `key : DistrTree.Key`. Every option I found, like `(← getMatchCore d e).1` are shielded behind some `private` declarations in core...

Reply:
We're only exercising this code interactively, so non-ideal performance is not a big concern.